doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/index.md

```diff
@@ -5,4 +5,383 @@ permalink: /:path/
+* The `ld` directory contains two linke scipts required for ESP8266 applications.
+* You should put generated Kaa C SDK tarball into `libs/kaa` directory and unpack it:
+
+        cd libs/kaa && tar zxf kaa-c*.tar.gz
+* `user/user_main.c` contains ESP8266 application entry ponit (`user_main()` function)
+and performs ESP8266-specific initizalizations (e.g. initialize UART).
+* `src/kaa_demo.c` is a platofrm-independent source file with minimal Kaa code.
+
+### Minimal code
+
+Each ESP8266 application starts its execution form `user_init()` function.
+Here we initialize UART and set it's baudrate to 115200 baud:
+
+~~~c
+    #include <freertos/FreeRTOS.h>
+    #include <freertos/task.h>
+
+    #include "uart.h"
+
+    void user_init(void)
+    {
+        uart_init_new();
+        UART_SetBaudrate(UART0, 115200);
+        UART_SetPrintPort(UART0);
+    }
+~~~
+
+Next, we should start a system task, since we run in FreeRTOS environment:
+
+~~~c
+    portBASE_TYPE error = xTaskCreate(main_task, (const signed char *)"main_task",
+            512, NULL, 2, NULL );
+    if (error < 0) {
+        printf("Error creating main_task! Error code: %ld\r\n", error);
+    }
+~~~
+
+In scope of this example, `main_task()` just calls `main()` function:
+
+~~~c
+    static void main_task(void *pvParameters)
+    {
+        (void)pvParameters;
+        main();
+        for(;;);
+    }
+~~~
+
+The `main()` function is defined in `src/kaa_demo.c` and starts minimal Kaa client using Kaa C SDK:
+
+~~~c
+    #include <stddef.h>
+    #include <kaa/kaa_error.h>
+    #include <kaa/kaa_context.h>
+    #include <kaa/platform/kaa_client.h>
+
+    static void loop_fn(void *context)
+    {
+        for(int i = 0;i<5;i++) {
+            printf("Hello, Kaa!\r\n");
+        }
+
+        kaa_client_stop(context);
+    }
+
+    int main(void)
+    {
+        printf("Initializing Kaa client\r\n");
+
+        kaa_client_t *kaa_client;
+
+        kaa_error_t error_code = kaa_client_create(&kaa_client, NULL);
+        if(error_code) {
+            printf("Failed to create Kaa client\r\n");
+            return 1;
+        }
+
+        error_code = kaa_client_start(kaa_client, loop_fn, (void*)kaa_client, 0);
+        if(error_code) {
+            printf("Failed to start Kaa main loop\r\n");
+            return 1;
+        }
+
+        kaa_client_destroy(kaa_client);
+        return 0;
+    }
+~~~
+
+### Build system overview
+
+The Kaa C SDK makes use of CMake build system and and though you can choose any build system of youir preference, it is recommended to use CMake for Kaa applications as well.
+This will make possible to tightly integrate your application's bulid system with Kaa SDK and use already provided ESP8266 toolchain file.
+
+#### CMakeLists.txt
+
+First, lets add Kaa SDK subdirectory, so that we can build it together with application.
+
+~~~CMake
+cmake_minimum_required(VERSION 3.0.2)
+
+project(kaa_demo C)
```

Comment:
These two lines has nothing to do with adding subdirectory, so describe them first separately (as a general preamble).